scheds/rust/scx_mitosis/src/bpf/mitosis.bpf.c

```diff
@@ -438,19 +442,28 @@ static inline int update_task_cell(struct task_struct *p, struct task_ctx *tctx,
 		/*
 		 * Cgroup lookup failed - this can happen during scheduler load
 		 * for tasks that were forked before the scheduler was loaded,
-		 * whose cgroups went offline before scx_cgroup_init() ran.
-		 * Only fall back to root cgroup if the workaround is enabled
-		 * and the task is exiting.
+		 * whose cgroups went offline before scx_cgroup_init() ran, or
+		 * when using tracepoint mode (cgroup_init not called and
+		 * cgroup_attach_task hasn't fired yet for this cgroup).
 		 */
-		if (exiting_task_workaround_enabled &&
-		    (p->flags & PF_EXITING)) {
+		if (use_cgroup_attach_tp) {
+			/* Fall back to root, or cell 0 if that fails */
+			struct cgroup *rootcg = READ_ONCE(root_cgrp);
+			if (rootcg)
+				cgc = lookup_cgrp_ctx_fallible(rootcg);
+			if (!cgc) {
+				tctx->cell		= 0;
+				tctx->configuration_seq = (u32)-1;
+				return update_task_cpumask(p, tctx);
+			}
+		} else if (exiting_task_workaround_enabled &&
+			   (p->flags & PF_EXITING)) {
```

Comment:
Hmm, is the only reason you need all this stuff because we won't get a cgroup_init for already created/attached cgroups on initialization? Maybe it would be cleaner to just iterate through all cgroups in mitosis_init and just initialize them explicitly?

Reply:
> Hmm, is the only reason you need all this stuff because we won't get a cgroup_init for already created/attached cgroups on initialization?

Yes once i figure out the minimal that works.

> Maybe it would be cleaner to just iterate through all cgroups in mitosis_init and just initialize them explicitly?

That would work, will figure it out, thx.